src/Symfony/Component/ClassLoader/ClassMapGenerator.php

```diff
@@ -118,6 +118,11 @@ private static function findClasses($path)
                 case T_CLASS:
                 case T_INTERFACE:
                 case SYMFONY_TRAIT:
+                    // Skip usage of ::CLASS constant
+                    if (0 !== $i && T_DOUBLE_COLON === $tokens[$i - 1][0]) {
```

Comment:
this does not work. You can have whitespaces between them, meaning it will not be the previous token. You can even have comments between them.

I suggest you to look at the PHP-CS-Fixer project, which implemented a way to find the previous meaningful token: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/02324909325ae387ceadfbe7a6955a3ebc1b0869/Symfony/CS/Tokenizer/Tokens.php#L748 (and related methods)